### ISSUE TITLE
🚨 [BUG] UnsupportedWarning: Status message is not supported by HTTP/2 (RFC7540 8.1.2.4)

### ISSUE THREAD

**Issue by JakiChen** [withastro/astro#12134]
### Astro Info

```block
Astro                    v4.15.11
Node                     v20.18.0
System                   macOS (x64)
Package Manager          yarn
Output                   server
Adapter                  vercel
Integrations             @storyblok/astro
                         @astrojs/sitemap
                         astro-robots
```

### Describe the Bug

When https is enabled locally —— **UnsupportedWarning: Status message is not supported by HTTP/2 (RFC7540 8.1.2.4)**

<img width="568" alt="截屏2024-10-06 23 31 36" src="https://github.com/user-attachments/assets/bd738d56-45ea-4b9b-abb9-9bfc4797323a">

### What's the expected result?

1. Make it clear whether this problem is a bug or real feedback, because this problem has not occurred in versions before 4.15.10.
2. I hope to share the data method of await fetch() that cannot access the endpoint when https is enabled.

> Currently, I can only deal with this problem by setting the dangerous behavior NODE_TLS_REJECT_UNAUTHORIZED to 0.

### Link to Minimal Reproducible Example

https://owainmavxv.github.stackblitz.io

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
I'm guessing it's something to do with https://github.com/withastro/astro/pull/12105

**Comment by JakiChen** [comment]
GitHub [repository](https://github.com/JakiChen/cloudflare_bug)

To enable local https, you must first install `config/certs/server.crt` to macOS `Keychain Access` and set `SSL` to `always allow`

Currently, this bug only occurs in Astro version 4.15.11, and versions before 4.15.10 to 4.15.4 are all normal (so it is not clear whether this is a bug or a problem caused by my configuration)

**Comment by JakiChen** [comment]
Heyyyy, @ascorbic 

## ` fix: don't set statusMessage in HTTP/2` 是这一个意思吗？还是说阉割了 HTTP/2 功能？
 
 在使用 HTTP/2 协议时，不能直接设置 statusMessage。在 HTTP/1.1 中，响应状态码（如 200）通常会有一个状态消息（如 OK）。但在 HTTP/2 协议中，状态消息被移除了，只有状态码，而没有 statusMessage。

如果你的项目在使用 HTTP/2 协议时尝试设置 statusMessage，可能会导致这个警告或错误。也就是说，这种设置在 HTTP/2 下是无效的，因为 HTTP/2 的设计简化了响应格式，只关注状态码，而不再传递状态消息。

这个提示不代表项目不支持 HTTP/2，而是说 不应该在 HTTP/2 请求中尝试设置 statusMessage。如果你遇到了这个问题，可以考虑调整你的代码，不再手动设置 statusMessage，这样就可以避免这个问题。

**Comment by JakiChen** [comment]
在 Astro@4.15.12 的最新版本中好像依旧没有被修复, 现在只能靠自己手动去`astro/dist/vite-plugin-astro-server/response.js(49:21)`删除这一行？`res.statusMessage = statusText;`

```js
async function writeWebResponse(res, webResponse) {
  const { status, headers, body, statusText } = webResponse;
  console.log(webResponse);
  const setCookieHeaders = Array.from(getSetCookiesFromResponse(webResponse));
  if (setCookieHeaders.length) {
    res.setHeader("set-cookie", setCookieHeaders);
  }
  const _headers = Object.fromEntries(headers.entries());
  if (headers.has("set-cookie")) {
    _headers["set-cookie"] = headers.getSetCookie();
  }
  console.log(res);
//   res.statusMessage = statusText;
  res.writeHead(status, _headers);
  if (body) {
    if (Symbol.for("astro.responseBody") in webResponse) {
      let stream = webResponse[Symbol.for("astro.responseBody")];
      for await (const chunk of stream) {
        res.write(chunk.toString());
      }
    } else if (body instanceof Readable) {
      body.pipe(res);
      return;
    } else if (typeof body === "string") {
      res.write(body);
    } else {
      const reader = body.getReader();
      res.on("close", () => {
        reader.cancel().catch(() => {
        });
      });
      while (true) {
        const { done, value } = await reader.read();
        if (done) break;
        if (value) {
          res.write(value);
        }
      }
    }
  }
  res.end();
}
```

不是很懂核心内部的代码核心，综合 ChatGPT 给的参考以及 console.log 出的结果应该是直接删除就好了（因为在HTTP/2中并不需要该值），希望 Astro 团队尽快修复，再次感谢 Astro 团队创造出如此容易上手理解的框架。

**Comment by ascorbic** [comment]
I merged it after `4.15.12` was released. It will be in `astro@4.15.13`, which hasn't been released yet.
